test/utils.ts

```diff
@@ -47,3 +49,98 @@ export function buildOperations(
     }
     return operations;
 }
+
+export function buildOperationAndDocument(
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    reducer: Reducer<any, any, any>,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    document: Document<any, any, any>,
+    action: Action,
+    index?: number
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+) {
+    const newDocument = reducer(document, action);
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-non-null-assertion
+    const operation = newDocument.operations[action.scope]
+        .slice()
+        .pop()! as Operation;
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+    return {
+        document: newDocument,
+        operation: {
+            ...operation,
+            index: index ?? operation.index
+        } as Operation<NOOPAction & Action>
+    };
+}
+
+export class BasicClient {
```

Comment:
now we can use this client to simulate how connect sends operations and sync with switchboard